src/EFCore.PG/Storage/Internal/NpgsqlDatabaseCreator.cs

```diff
@@ -187,6 +187,19 @@ await unpooledRelationalConnection.OpenAsync(errorsExpected: true, cancellationT
             // Pretty awful hack around #104
             return false;
         }
+        catch (NpgsqlException e) when (
+            // This happens when Npgsql attempts to connect to multiple hosts.
+            e.InnerException is AggregateException &&
```

Comment:
@Brar, as per our offline discussion, can you change this to return "does not exist" if **any** of the exceptions within the AggregateException is "does not exist"?